File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int c_letters(string text);
int c_words(string text);
int c_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");

    printf("%i and %i and %i\n", c_letters(text), c_words(text), c_sentences(text));

    float L = (float) c_letters(text) / c_words(text) * 100;
    float S = (float) c_sentences(text) / c_words(text) * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    printf("%f so %f\n", L, S);

    if (index < 1)
    {
        printf("Before grade 1\n");
        return 0;
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
        return 0;
    }

    printf("Grade %i\n", index);
    return 0;
}

int c_letters(string text)
{
    int letters = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int c_words(string text)
{
    int words = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]) && (isspace(text[i + 1]) || ispunct(text[i + 1])))
        {
            if (text[i + 1] == '-')
            {
                continue;
            }
            words++;
        }
    }
    return words;
}

int c_sentences(string text)
{
    int sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}

File: sentimental-readability/readability.py
from cs50 import get_string
import string


def main():
    text = get_string("Text: ")

    L = c_letters(text) / c_words(text) * 100
    S = c_sentences(text) / c_words(text) * 100
    index = round(0.0588 * L - 0.296 * S - 15.8)

    if index < 1:
        return print("Before Grade 1\n")
    elif index > 16:
        return print("Grade 16+\n")
    else:
        return print(f"Grade {index}")


def c_letters(txt):
    letters = 0
    for char in txt:
        if char.isalpha():
            letters += 1
    return letters


def c_words(txt):
    spaces = 0
    for char in txt:
        if (char.isspace()):
            spaces += 1
    words = spaces + 1
    return words


def c_sentences(txt):
    sentences = 0
    for i in range(len(txt) - 1):
        if (txt[i].isspace() and txt[i-1] in ".!?"):
            sentences += 1
    sentences += 1
    return sentences


if __name__ == "__main__":
    main()
